_studio/shared/umc/codec/h264_dec/src/umc_h264_va_supplier.cpp

```diff
@@ -327,40 +327,8 @@ Status VATaskSupplier::AddSource(MediaData * pSource)
     if (!pSource)
         return MFXTaskSupplier::AddSource(pSource);
 
-    notifier0<LazyCopier> memory_leak_preventing_slice(&m_lazyCopier, &LazyCopier::CopyAll);
-
-    uint32_t const flags = pSource->GetFlags();
-    if (flags & MediaData::FLAG_VIDEO_DATA_NOT_FULL_FRAME)
-        return MFXTaskSupplier::AddSource(pSource);
-
-    if (m_currentView == INVALID_VIEW_ID)
-        return MFXTaskSupplier::AddSource(pSource);
-
-    ViewItem &view = GetView(m_currentView);
-    if (view.pCurFrame && view.pCurFrame->m_PictureStructureForDec < FRM_STRUCTURE)
-    {
-        H264Slice* pFirstFrameSlice = view.pCurFrame->GetAU(0)->GetSlice(0);
-        //we need to check only for first slice
-        if (pFirstFrameSlice)
-            return MFXTaskSupplier::AddSource(pSource);
-    }
-
-    H264DBPList* pDPB = view.GetDPBList(0);
-    VM_ASSERT(pDPB && "DPB pointer should be valid here");
-    if (!pDPB)
-        return UMC_ERR_FAILED;
-
-    //check if we have free frame
-    if (pDPB->countAllFrames() < view.maxDecFrameBuffering + m_DPBSizeEx ||
-        pDPB->IsDisposableExist())
-        return MFXTaskSupplier::AddSource(pSource);
-
-    H264DecoderFrame* completed = 0;
-    Status umcRes = CompleteDecodedFrames(&completed);
-    if (umcRes != UMC_OK)
-        return pSource || !completed ? umcRes : UMC_OK;
-
-    return UMC_WRN_INFO_NOT_READY;
+    notifier0<LazyCopier> copy_slice_data(&m_lazyCopier, &LazyCopier::CopyAll);
```

Comment:
so this is:
```
if (pSource)
    notifier0<LazyCopier> copy_slice_data(&m_lazyCopier, &LazyCopier::CopyAll);

return MFXTaskSupplier::AddSource(pSource);
```

Reply:
No it isn't. In your case [LazyCopier] will always (when pSource != NULL) copy slice data while we needn't do this when full frame was submitted.